pkg/loopback/loopback_test.go

```diff
@@ -0,0 +1,136 @@
+// +build linux,cgo
+
+package loopback
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+	"syscall"
+	"testing"
+
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const filePath = "/tmp/"
+
+var rootEnabled bool
+
+func init() {
+	flag.BoolVar(&rootEnabled, "test.root", false, "enable tests that require root")
```

Comment:
I agree that if a test is potentially destructive it should be opt in. I'm still not clear about what exactly is destructive or not safe about these tests, but let's say that they are.

There are other options for opt-in that I think are better suited for a test suite. Build tags are good if the entire file should be ignored, and environment variables are great for individual tests.

I think the advantage to these options is that you don't have to special case testing of any package. You can set the build tag or environment variable for everything, and not  worry about which packages happen to be selected by the user on a specific test run.

> As it is, they take way to long to run

Yes they do, but the unit tests are not really a significant part of that time. Most of the time is running way too many integration tests.